tastytrade/streamer.py

```diff
@@ -561,6 +586,387 @@ async def _map_message(self, message) -> None:
+        else:
+            raise TastytradeError(f'Unknown message type received: {message}')
+
+
+class DataStreamerNew(DataStreamer):
+    """
+    A :class:`DataStreamer` object is used to fetch quotes or greeks
+    for a given symbol or list of symbols. It should always be
+    initialized using the :meth:`create` function, since the object
+    cannot be fully instantiated without using async.
+
+    Example usage::
+
+        from tastytrade import DataStreamer
+        from tastytrade.dxfeed import EventType
+
+        # must be a production session
+        streamer = await DataStreamer.create(session)
+
+        subs = ['SPY', 'GLD']  # list of quotes to fetch
+        await streamer.subscribe(EventType.QUOTE, subs)
+        quotes = []
+        async for quote in streamer.listen(EventType.QUOTE):
+            quotes.append(quote)
+            if len(quotes) >= len(subs):
+                break
+
+    """
+
+    def __init__(self, session: ProductionSession):
+        self._counter = 0
+        self._lock: Lock = Lock()
+        self._queues: Dict[str, Queue] = {
+            EventType.CANDLE: Queue(),
+            EventType.GREEKS: Queue(),
+            EventType.PROFILE: Queue(),
+            EventType.QUOTE: Queue(),
+            EventType.SUMMARY: Queue(),
+            EventType.THEO_PRICE: Queue(),
+            EventType.TIME_AND_SALE: Queue(),
+            EventType.TRADE: Queue(),
+            EventType.UNDERLYING: Queue()
+        }
+        self._queue_channels: Dict[str, int] = {
+            EventType.CANDLE: 1,
+            EventType.GREEKS: 3,
+            EventType.PROFILE: 5,
+            EventType.QUOTE: 7,
+            EventType.SUMMARY: 9,
+            EventType.THEO_PRICE: 11,
+            EventType.TIME_AND_SALE: 13,
+            EventType.TRADE: 15,
+            EventType.UNDERLYING: 17
+        }
+        self._subscription_state: Dict[int, str] = {
+            self._queue_channels[EventType.CANDLE]: "CHANNEL_CLOSED",
+            self._queue_channels[EventType.GREEKS]: "CHANNEL_CLOSED",
+            self._queue_channels[EventType.PROFILE]: "CHANNEL_CLOSED",
+            self._queue_channels[EventType.QUOTE]: "CHANNEL_CLOSED",
+            self._queue_channels[EventType.SUMMARY]: "CHANNEL_CLOSED",
+            self._queue_channels[EventType.THEO_PRICE]: "CHANNEL_CLOSED",
+            self._queue_channels[EventType.TIME_AND_SALE]: "CHANNEL_CLOSED",
+            self._queue_channels[EventType.TRADE]: "CHANNEL_CLOSED",
+            self._queue_channels[EventType.UNDERLYING]: "CHANNEL_CLOSED"
+        }
+
+        #: The unique client identifier received from the server
+        self._session = session
+        self._authenticated = False
+        api_token = self.get_api_token()
+        self._auth_token = api_token['token']
+        self._wss_url = api_token['dxlink-url']
+
+        self._connect_task_new = asyncio.create_task(self._connect())
+        self._keepalive = asyncio.create_task(self._keepalive())
```

Comment:
The errors are because you're using `_keepalive` as the name of the method and the variable.